File: languages/js/src/entity/identity/issuer.ts
import { Key } from "../key/key";
import { Did } from "./did";
import { DidMethod } from "./did_method";

/**
 * represents a Issuer identity.
 */
export class Issuer {
    did: Did;
    key: Key;

    /**
     * Constructs a Issuer object with the specified parameters.
     * @param did 
     * @param didMethod 
     */
    constructor(did: string, key: Key, didMethod: DidMethod) {
        this.did = new Did(did, didMethod);
        this.key = key;
    }
}

File: languages/go/entity/identity/issuer.go
package identity

import (
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

// Issuer represents an Issuer identity.
type Issuer struct {
	Did Did
	Key key.Key
}

// NewIssuer returns a new instance of Issuer identity for the given parameters.
func NewIssuer(did string, didMethod DidMethod, key key.Key) Issuer {
	return Issuer{
		Did: NewDid(did, didMethod),
		Key: key,
	}
}
